### ISSUE TITLE
Cannot launch metadata service with provided image

### ISSUE THREAD

**Issue by madison-ookla** [stemma-ai/amundsen-custom#7]
I have tried using the provided docker images to run the service, but I'm encountering the following error with the metadata service:

<details><summary> Error message</summary>

```
Traceback (most recent call last):
  File "/usr/local/bin/gunicorn", line 8, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 58, in run
    WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 228, in run
    super().run()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 72, in run
    Arbiter(self).run()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 58, in __init__
    self.setup(app)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 118, in setup
    self.app.wsgi()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 49, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.7/site-packages/gunicorn/util.py", line 358, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/amundsen/metadata/wsgi.py", line 1, in <module>
    from metadata_service import create_app
  File "/usr/local/lib/python3.7/site-packages/metadata_service/__init__.py", line 17, in <module>
    from metadata_service.api.badge import BadgeAPI
  File "/usr/local/lib/python3.7/site-packages/metadata_service/api/__init__.py", line 10, in <module>
    from metadata_service.proxy import BaseProxy
  File "/usr/local/lib/python3.7/site-packages/metadata_service/proxy/__init__.py", line 10, in <module>
    from metadata_service.proxy.base_proxy import BaseProxy
  File "/usr/local/lib/python3.7/site-packages/metadata_service/proxy/base_proxy.py", line 13, in <module>
    from metadata_service.entity.dashboard_detail import \
  File "/usr/local/lib/python3.7/site-packages/metadata_service/entity/dashboard_detail.py", line 41, in <module>
    class DashboardSchema(AttrsSchema):
  File "/usr/local/lib/python3.7/site-packages/marshmallow/schema.py", line 124, in __new__
    dict_cls=dict_cls,
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/scheme.py", line 110, in get_declared_fields
    fields.update(converter.convert_all(target, ignore, klass.opts.field_configs))
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/ext/attrs.py", line 49, in convert_all
    return super().convert_all(target, ignore, configs)
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/converter.py", line 80, in convert_all
    for k, v in self._get_type_hints(target, ignore)
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/converter.py", line 80, in <dictcomp>
    for k, v in self._get_type_hints(target, ignore)
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/converter.py", line 66, in convert
    typehint, opts, field_name=field_name, target=target
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/converter.py", line 117, in _field_from_typehint
    return field_constructor(self, subtypes, kwargs)
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/registry.py", line 53, in _list_converter
    if converter.is_scheme(subtypes[0]):
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/converter.py", line 84, in is_scheme
    constructor = self.registry.get(typehint)
  File "/usr/local/lib/python3.7/site-packages/marshmallow_annotations/registry.py", line 120, in get
    raise AnnotationConversionError(f"No field factory found for {target!r}")
marshmallow_annotations.exceptions.AnnotationConversionError: No field factory found for <class 'amundsen_common.models.user.User'>
```

</details>

This fails when I set the upstream commit for the metadata service to the v3.3.0 tag (which I also have to modify the Dockerfile for due to the requirements not being explicitly installed in that version) and when I point the upstream to master.

**Comment by dorianj** [comment]
Hmm, weird, I haven't remotely seen this before. 

I wonder if the fix in https://github.com/amundsen-io/amundsenmetadatalibrary/pull/286/files would resolve this? I know there's a mess brewing w.r.t. marshmallow.

@verdan do you recognize this?

**Comment by madison-ookla** [comment]
Wow, talk about timing! Using the most recent version on master (v3.4.0) worked!! Thank you 😄  I can try and get a PR out to update this project to point at that version.

**Comment by verdan** [comment]
Super, would appreciate that. They cloned and pushed a fork of marshmallow annotations to PyPi yesterday. That fixed this issue.

**Comment by madison-ookla** [comment]
Closed via #8
